### PR TITLE
Tforkner/3100 move apds to mongodb

### PR DESCRIPTION
**Pull request by thetif**
**Description-**
Moves APDs from postgres to mongodb

**This pull request changes...**

- adds mongo to build
- everything in the system that was querying postgres for APDs
- changes a little about the data structure of APDs to make it stricter

### This pull request is ready to review when...

- [x] Automated tests are updated (and all tests are passing)
- [x] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented

### This pull request can be merged when…

- [ ] Code has been reviewed by someone other than the original author
- [ ] QA has verified the accessibility and functionality related to the change
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

**Review by tbolt — APPROVED** [review]
Code looks good. very nice

File: integrationTests/cypress/integration/admin/02-state-admin-letters.spec.js [hunk 1]
```diff
@@ -42,7 +42,7 @@ describe('state admin letters table', { tags: ['@fed', '@admin'] }, () => {
     cy.get('#state-admin-letters').contains('Rows per page:');
   });
 
-  it('allows filtering by status', () => {
+  xit('allows filtering by status', () => {
```
[the remark this answers is not in the document]
Comment (thetif) [reply integrationTests/cypress/integration/admin/02-state-admin-letters.spec.js:45]: oh maybe it was commented out in a previous version and just didn't get merged back in, I'll check it

File: api/routes/apds/post.test.js [hunk 2]
```diff
@@ -66,30 +71,30 @@ tap.test('apds POST endpoint', async endpointTest => {
...
-                params: { type: 'string' },
-                message: 'must be string'
-              }
-            ]
-          }),
-          'HTTP status set to 400'
-        );
-      }
-    );
+    // tests.test(
```
[the remark this answers is not in the document]
Comment (thetif) [reply api/routes/apds/post.test.js:74]: I'll delete that

File: packer/mongo.sh [hunk 3]
```diff
@@ -0,0 +1,303 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (tbolt) [reply packer/mongo.sh:1]: thanks, that makes a ton of sense

**Comment by jeromeleecms** [comment]
Functionally, I can't tell the difference between using the app to build/store APDs with this in our old data store :tada: -- how do we check/verify that the existing data was migrated properly?

**Comment by thetif** [comment]
@jeromeleecms I have asked Bill to set up a jumpbox so I can access the databases, then I'll be able to verify that everything came over. Also, the migration is non-destructive to the postgres APDs so if we miss something we can get it again.

**Review by jeromeleecms — APPROVED** [review]
Seems to be working; Plan in place to verify all things are moving over.

**Pull request merged**